### DISCUSSION TITLE
Help on router

### DISCUSSION THREAD

**Discussion by raghavan97** [bacnet-stack/bacnet-stack#579] [General]
Hi,

I am trying to use the "router" application to route between 2 networks **192.168.12.*** *(network 31)* to **192.168.1.*** *(network 32)*

I have an interface enp7s0 (192.168.12.22)on network 31 and enp6s0(192.168.1.12) on network 32

I created a r.cfg file with the following content

    ports =
    (
    	{
    			device_type = "bip";
    			device = "enp7s0";
    			port = 47808;
    			network = 31;
    	},
    	{
    			device_type = "bip";
    			device = "enp6s0";
    			port = 47808;
    			network = 32;
    	}
    );

The following env variables were exported
**BACNET_IFACE=enp7s0
BACNET_IP_NET=31**

I am running the router app as follows

    myapp:~/code/bacnet-stack/bin$ ./router -c ./r.cfg
    opt = c
    dev_type = bip
    dev_type = bip
    cmd file parse success
    Initializing...
    Interface: enp7s0
    Interface: enp6s0
    IP Address: 192.168.1.12
    IP Broadcast Address: 192.168.1.255
    UDP Port: 0xBAC0 [47808]
    IP Address: 192.168.10.22
    IP Broadcast Address: 192.168.10.255
    UDP Port: 0xBAC0 [47808]

There is a bacnet device on 192.168.1.2 which I expect to be seen in the BACnet explorer.

I am using a BACnet explorer that is sending out who is enp7s0, but it is not getting any response from the router app

Just to confirm if my IP setup is right, I ran 

    ./bacserv 333 boo 

and that is being detected by the BACnet Explorer. I see the device boo on the BACnet Explorer.

Can you please tell where I am going wrong ?

I am using the latest from the repository.

**Comment by raghavan97** [comment]
Hi All,

I figured out the issue. The Makefile in apps/router directory does a 

    CFLAGS += -I${SOURCE_DIR} -I${BACNET_PORT_DIR}

The CFLAGS += causes a bunch of Warning options (harmless) and also a bunch of Defines to get enabled. Those -DXXX are causing it fail. (See below)

I changed it to 

    CFLAGS =  -I${SOURCE_DIR} -I${BACNET_PORT_DIR}

This fixes the issue

These are the additional options coming up during the compilation. 

    -Wall -Wextra -Wfloat-equal -Wconversion -Wparentheses -pedantic -Wunused-parameter -Wunused-variable -Wreturn-type -Wunused-function -Wreturn-type -Wunused-value -Wswitch-default -Wuninitialized -Winit-self -Wno-sign-conversion -Wno-conversion -Wno-sign-compare -Wno-long-long -Wno-implicit-fallthrough -Wno-attributes  -Os  -DBACDL_BIP=1 -DBBMD_ENABLED=1 -DBBMD_CLIENT_ENABLED -DWEAK_FUNC=  -DPRINT_ENABLED=1 -DBACAPP_ALL -DBACFILE -DINTRINSIC_REPORTING -DBACNET_TIME_MASTER -DBACNET_PROPERTY_LISTS=1 -DBACNET_PROTOCOL_REVISION=24 -I/home/rags/code/bacnet-stack/src -I/home/rags/code/bacnet-stack/ports/linux -std=gnu89 -ffunction-sections

@skarg , You might have a better idea on which -D is causing the issue.

**Reply by skarg** [reply]
Fixed makefile for building the Linux router application with pull request #585. The pull request fixes #579.
